src/graphql.tsx

```diff
@@ -475,15 +513,16 @@ export default function graphql(
           return renderedElement;
         }
 
-        if (operationOptions.withRef) mergedPropsAndData.ref = 'wrappedInstance';
+        if (operationOptions.withRef) mergedPropsAndData['ref'] = 'wrappedInstance';
 
-        this.renderedElement = createElement(WrappedComponent, mergedPropsAndData);
+        this.renderedElement = createElement(WrappedComponent as React.ComponentClass<T>, mergedPropsAndData);
         return this.renderedElement;
       }
     }
 
     // Make sure we preserve any custom statics on the original component.
-    return hoistNonReactStatics(GraphQL, WrappedComponent, {});
-  };
+    hoistNonReactStatics(GraphQL, WrappedComponent, {});
 
+    return GraphQL as typeof WrappedComponent;
```

Comment:
Two issues:

1. Since we currently cannot infer the types of the props that are decorated, I am simply using the type of the wrapped component here. Obviously, this gives us nothing w/r/t the types that are passed in; but as it stands the user was not getting any of this information to begin with.

2. I cannot return GraphQL directly since the type is not exported. And I do not know how to export the type :confused: